app/services/external_http_service.rb

```diff
@@ -0,0 +1,91 @@
+class ExternalHTTPService
+
+  require 'httparty'
+
+  SUCCESS_CODES = [200, 201, 202].freeze
+
+
+  # Basic method for getting a response from some external service
+  # via HTTParty
+  #
+  # subclasses can just override the 'get_ ...' methods or they can
+  # override this method completely
+  def self.get_response(*args)
+
+    response = HTTParty.get(response_url(args),
```

Comment:
I think it would be clearer to call this `request_url` rather than `response_url`.